### PR TITLE
gh-83004: Harden _socket init

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-83004

Automerge-Triggered-By: GH:erlend-aasland

### CODE REVIEW

File: Modules/socketmodule.c [hunk 1]
```diff
@@ -7338,1394 +7338,1427 @@ static struct PyModuleDef socketmodule = {
...
-    if (socket_herror == NULL)
-        return NULL;
-    PyModule_AddObject(m, "herror", Py_NewRef(socket_herror));
-    socket_gaierror = PyErr_NewException("socket.gaierror", PyExc_OSError,
-        NULL);
-    if (socket_gaierror == NULL)
-        return NULL;
-    PyModule_AddObject(m, "gaierror", Py_NewRef(socket_gaierror));
-    PyModule_AddObjectRef(m, "timeout", PyExc_TimeoutError);
+#define ADD_EXC(MOD, NAME, VAR) do {                                \
```
Comment (hauntsaninja) [inline Modules/socketmodule.c]: nit: could call this ADD_OSERROR_EXC?

**Review by hauntsaninja — APPROVED** [review]
This looks good to me, thank you!

File: Modules/socketmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/socketmodule.c]: Yeah, I thought about making the naming more explicit. I'll consider your suggestion before merging 🙂 Thanks for the review!

Comment (erlend-aasland) [reply Modules/socketmodule.c]: I added base as an argument to the macro; it is better to be explicit.

**Pull request merged**